packages/office-ui-fabric-react/src/utilities/decorators/withResponsiveMode.tsx

```diff
@@ -40,74 +51,87 @@ export function setResponsiveMode(responsiveMode: ResponsiveMode | undefined): v
 export function withResponsiveMode<TProps extends { responsiveMode?: ResponsiveMode }, TState>(
   ComposedComponent: new (props: TProps, ...args: any[]) => React.Component<TProps, TState>
 ): any {
-  const resultClass = class WithResponsiveMode extends BaseDecorator<TProps, IWithResponsiveModeState> {
-    constructor(props: TProps) {
-      super(props);
-      this._updateComposedComponentRef = this._updateComposedComponentRef.bind(this);
-
-      this.state = {
-        responsiveMode: _defaultMode || _lastMode || ResponsiveMode.large
-      };
-    }
+  // tslint:disable-next-line:function-name
+  function WithResponsiveModeBase(props: TProps, forwardedRef: React.Ref<React.Component<TProps, TState>>): JSX.Element | null {
+    const [responsiveMode = ResponsiveMode.unknown, setCurrentResponsiveMode] = React.useState<ResponsiveMode>(
+      _defaultMode || _lastMode || ResponsiveMode.large
+    );
 
-    public componentDidMount(): void {
-      this._events.on(window, 'resize', this._onResize);
-      this._onResize();
-    }
+    const rootRef = React.useRef<HTMLDivElement | null>(null);
 
-    public componentWillUnmount(): void {
-      this._events.dispose();
-    }
+    React.useLayoutEffect(() => {
+      if (typeof window !== 'undefined') {
+        const events = new EventGroup(null);
 
-    public render(): JSX.Element | null {
-      const { responsiveMode } = this.state;
+        const onResize = () => {
+          const element = rootRef.current;
+          const currentWindow = (element && getWindow(element)) || window;
+
+          if (currentWindow) {
+            setCurrentResponsiveMode(getResponsiveMode(currentWindow));
+          }
+        };
 
+        onResize();
+
+        // There is not a good way to subscribe to the component instance window.
+        events.on(window, 'resize', onResize);
+
+        return () => {
+          events.dispose();
+        };
+      }
+    }, []);
+
+    const componentElement = React.useMemo(() => {
       return responsiveMode === ResponsiveMode.unknown ? null : (
-        <ComposedComponent ref={this._updateComposedComponentRef} responsiveMode={responsiveMode} {...(this.props as any)} />
+        <>
+          <ComposedComponent {...(props as any)} ref={forwardedRef} responsiveMode={responsiveMode} />
+          <div className={css('ms-withResponsiveMode', WithResponsiveModeStyles.root)} ref={rootRef} />
```

Comment:
Why are you adding a class name?

Reply:
This way its clear why there area bunch of extra `div` elements across all the snapshots.